src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryAPI.groovy

```diff
@@ -117,12 +117,20 @@ abstract class ArtifactoryAPI {
      * Converts the provided base name (expected to be a GitHub repository name of the form 'org/name') to a user name
      * for a non-existing token user.
      *
+     * Artifactory has a limit of 58 characters for username
+     *
      * @link https://www.jfrog.com/confluence/display/JFROG/Access+Tokens#AccessTokens-SupportAuthenticationforNon-ExistingUsers
      * @param baseName
      * @return
      */
     static @NonNull String toTokenUsername(String baseName) {
-        return 'CD-for-' + baseName.replaceAll('[ /]', '__')
+        def name = 'CD-for-' + baseName.replaceAll('[ /]', '__')
+        if (name.length() > 58) {
+            // Artifactory has an undocumented max length for usernames of 58 chars (and possibly other types) although documented to be 255 characters
+            // If length is exceeded, use 58 chars of the prefix+name, separator, and 8 hopefully unique chars (prefix of name's SHA-256)
+            name = name.substring(0, 50) + '_' + sha256(name).substring(0, 7)
```

Comment:
I think the comment and code don't match here. This would make the comment right, but maybe you really do want 8 characters of the hash and one fewer prefix character:

```suggestion
            // If length is exceeded, use 50 chars of the prefix+name, separator, and 7 hopefully unique chars (prefix of name's SHA-256)
            name = name.substring(0, 50) + '_' + sha256(name).substring(0, 7)
```


The comparable code in `toGeneratedName` is also wrong and doesn't match its comment (substring `endIndex` is exclusive). It uses 54 + 1 + 7 = 62 characters, so maybe the intended code was `name.substring(0, 55) + '_' + sha256(name).substring(0, 8)`

Reply:
Absolutely right